src/test/java/teammates/common/datatransfer/questions/FeedbackRankOptionsQuestionDetailsTest.java

```diff
@@ -20,4 +34,172 @@ public void testIsFeedbackParticipantCommentsOnResponsesAllowed_shouldReturnFals
+        feedbackQuestionDetails.setMinOptionsToBeRanked(3);
+        feedbackQuestionDetails.setMaxOptionsToBeRanked(5);
+        feedbackQuestionDetails.setOptions(Arrays.asList("1", "2", "3", "4", "5", "6"));
+        List<String> errorResponse = new ArrayList<>();
+
+        FeedbackRankOptionsResponseDetails feedbackResponseDetails = new FeedbackRankOptionsResponseDetails();
+        feedbackResponseDetails.setAnswers(Arrays.asList(1, 2));
+        errorResponse.add("You must rank at least " + feedbackQuestionDetails.minOptionsToBeRanked + " options.");
+        assertEquals(errorResponse,
+                feedbackQuestionDetails.validateResponsesDetails(Arrays.asList(feedbackResponseDetails), 1));
+        errorResponse.clear();
+
+        feedbackResponseDetails.setAnswers(Arrays.asList(1, 2, 3, 4, 5, 6));
+        errorResponse.add("You can rank at most " + feedbackQuestionDetails.maxOptionsToBeRanked + " options.");
+        assertEquals(errorResponse,
+                feedbackQuestionDetails.validateResponsesDetails(Arrays.asList(feedbackResponseDetails), 1));
+    }
+
+    @Test
+    public void testValidateResponseDetails_invalidRankOption_errorReturned() {
+        FeedbackRankOptionsQuestionDetails feedbackQuestionDetails = new FeedbackRankOptionsQuestionDetails();
+        feedbackQuestionDetails.setOptions(Arrays.asList("1", "2", "3"));
+        List<String> errorResponse = new ArrayList<>();
+
+        FeedbackRankOptionsResponseDetails feedbackResponseDetails = new FeedbackRankOptionsResponseDetails();
+        feedbackResponseDetails.setAnswers(Arrays.asList(0));
+        errorResponse.add(FeedbackRankOptionsQuestionDetails.ERROR_INVALID_RANK_RESPONSE);
+        assertEquals(errorResponse,
+                feedbackQuestionDetails.validateResponsesDetails(Arrays.asList(feedbackResponseDetails), 1));
+
+        feedbackResponseDetails.setAnswers(Arrays.asList(5));
+        assertEquals(errorResponse,
+                feedbackQuestionDetails.validateResponsesDetails(Arrays.asList(feedbackResponseDetails), 1));
+    }
+
+    @Test
+    public void testValidateResponseDetails_validRankOption_noError() {
+        FeedbackRankOptionsQuestionDetails feedbackQuestionDetails = new FeedbackRankOptionsQuestionDetails();
+        feedbackQuestionDetails.setOptions(Arrays.asList("1", "2", "3", "4", "5"));
+
+        FeedbackRankOptionsResponseDetails feedbackResponseDetails1 = new FeedbackRankOptionsResponseDetails();
+        feedbackResponseDetails1.setAnswers(Arrays.asList(2, 3));
+        FeedbackRankOptionsResponseDetails feedbackResponseDetails2 = new FeedbackRankOptionsResponseDetails();
+        feedbackResponseDetails2.setAnswers(Arrays.asList(3, 5));
+        assertTrue(feedbackQuestionDetails.validateResponsesDetails(
+                Arrays.asList(
+                        feedbackResponseDetails1,
+                        feedbackResponseDetails2),
+                1).isEmpty());
```

Comment:
I think it'd be better to have the closing bracket on a new line, would be easy to miss the value `1` here